packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift

```diff
@@ -832,7 +854,16 @@ public final class FileDownloader {
 
       successBlock(data, response)
     }
-    task.resume()
+
+    if let progressBlock = progressBlock, let task = task {
+      progressObservationsLock.lock()
+      progressObservations[task] = task.progress.observe(\.fractionCompleted) { progress, _ in
```

Comment:
i have some mixed feelings of using observers instead of session delegate/await bytes() approach. The observers get cleared at appropriate places so i think this is much simpler, but let me know if I should refactor to delegate or the bytes API 🙏

Reply:
do you know how does ios report `fractionCompleted` progress when no content-length header?